web/src/components/tasks/search-business-name/SearchBusinessNameForm.tsx

```diff
@@ -51,42 +48,128 @@ export const SearchBusinessNameForm = (props: Props): ReactElement => {
 
   const { Config } = useConfig();
   const didInitialSearch = useRef<boolean>(false);
-  const { userData } = useUserData();
-  const [nameAvailability, setNameAvailability] = useState<NameAvailability | undefined>(undefined);
-  const [submittedName, setSubmittedName] = useState<string>("");
+  const { userData, updateQueue } = useUserData();
+  const {
+    setFormationFormData,
+    setBusinessNameAvailability,
+    setDbaBusinessNameAvailability,
+    setFieldsInteracted,
+  } = useContext(BusinessFormationContext);
+
+  const setNameAvailability = props.isDba ? setDbaBusinessNameAvailability : setBusinessNameAvailability;
 
+  const FIELD_NAME = "businessName";
   const SearchBusinessNameErrorLookup: Record<SearchBusinessNameError, string> = {
     BAD_INPUT: Config.searchBusinessNameTask.errorTextBadInput,
     SEARCH_FAILED: Config.searchBusinessNameTask.errorTextSearchFailed,
   };
 
-  useEffect(() => {
-    props.onChange && props.onChange(nameAvailability);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [nameAvailability]);
+  const emptyNameAvailability: NameAvailability = {
+    similarNames: [],
+    status: undefined,
+    lastUpdatedTimeStamp: "",
+  };
 
   const Unavailable = props.unavailable;
   const Available = props.available;
-  const onSubmit = props.onSubmit;
+
+  const handleBusinessNameAvailability = async (
+    nameAvailability: NameAvailability,
+    submittedName: string
+  ): Promise<void> => {
+    const validName = ["AVAILABLE", "UNAVAILABLE"].includes(nameAvailability.status ?? "");
+    const nameUnavailable = nameAvailability.status === "UNAVAILABLE";
+
+    if (updateQueue && validName) {
+      const needsNexusDbaName = nameUnavailable ? true : false;
+      setFieldsInteracted([FIELD_NAME]);
+      await updateQueue
+        .queueFormationData(
+          {
+            businessNameAvailability: nameAvailability,
+          },
+          { businessName: submittedName }
+        )
+        .queueProfileData({
+          businessName: submittedName,
+          nexusDbaName: emptyProfileData.nexusDbaName,
+          needsNexusDbaName,
+        })
+        .update();
+
+      setFormationFormData((previousFormationData) => {
+        return {
+          ...previousFormationData,
+          businessName: submittedName,
+        };
+      });
+    }
+  };
+
+  const handleDbaBusinessNameAvailability = async (
+    nameAvailability: NameAvailability,
+    submittedName: string
+  ): Promise<void> => {
+    if (updateQueue && nameAvailability.status === "AVAILABLE") {
```

Comment:
Is this right? We're only updating/saving the name availablity if available? Shouldn't we save it if it comes back unavailable also?

I'm a little confused by the "simplification"/breakdown of these two `handle` functions - these weren't here before, right? unfortunately I can't see the old version

Reply:
also, same note about guard clause for update queue

Reply:
They were not separated before. It was all one thing. I was initially keeping it all together to make it easier for myself while I was considering the next part of the refactor. I just opted to break it out now.

The current behavior is that we don't want to save unavailable DBA name searches, and for biz name we don't want to save anything that isn't unavailable or available.